pennylane/measurements/expval.py

```diff
@@ -114,7 +127,11 @@ def process_samples(
         # estimate the ev
         op = self.mv if self.mv is not None else self.obs
         with qml.queuing.QueuingManager.stop_recording():
-            samples = qml.sample(op=op).process_samples(
+            samples = qml.sample(
+                op=op,
+                eigvals=self._eigvals,
+                wires=self.wires if self._eigvals is not None else None,
```

Comment:
Very minor, take it or leave it - any valid input for eigvals is going to be a `list` or `array`, and therefore truthy, right? So it can just be:

```suggestion
                wires=self.wires if self._eigvals else None,
```

Reply:
For some reason I get an error with just `if self._eigvals`:

```
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```